tests/aggregation/tests.py

```diff
@@ -2102,13 +2102,40 @@ def test_exists_extra_where_with_aggregate(self):
         )
         self.assertEqual(len(qs), 6)
 
-    def test_aggregation_over_annotation_shared_alias(self):
-        self.assertEqual(
-            Publisher.objects.annotate(agg=Count("book__authors")).aggregate(
-                agg=Count("agg"),
+    def test_referenced_aggregation_over_annotation(self):
+        total_books_qs = (
+            Author.book_set.through.objects.values("author")
+            .filter(author=OuterRef("pk"))
+            .annotate(total=Count("book"))
+        )
+        annotated_authors = Author.objects.annotate(
+            total_books=Subquery(total_books_qs.values("total"))
+        ).values("pk", "total_books")
+        totals = annotated_authors.aggregate(
+            sum_total_books=Sum("total_books"),
+            sum_total_books_not_null=Case(
+                When(sum_total_books__isnull=True, then=0),
+                default=Sum("total_books"),
+                output_field=IntegerField(),
+            ),
+        )
+        self.assertEqual(totals["sum_total_books"], 10)
+        self.assertEqual(totals["sum_total_books_not_null"], 10)
+
+    def test_referenced_aggregation_over_aggregation(self):
+        annotated_authors = Author.objects.annotate(
+            total_books=Count("book"),
+        ).values("pk", "total_books")
+        totals = annotated_authors.aggregate(
+            sum_total_books=Sum("total_books"),
+            sum_total_books_not_null=Case(
+                When(sum_total_books__isnull=True, then=0),
+                default=Sum("total_books"),
+                output_field=IntegerField(),
             ),
-            {"agg": 5},
         )
+        self.assertEqual(totals["sum_total_books"], 10)
+        self.assertEqual(totals["sum_total_books_not_null"], 10)
```

Comment:
Possibly a redundant test. It might be worth adding one that uses simple self-referencing aggregates while taking the optimized path introduced in 59bea9efd2768102fc9d3aedda469502c218e9b7

e.g.

```python
Author.objects.aggregate(
    cnt=Count("*"),
    coalesced_cnt=Coalesce("cnt", 0)
)
```